app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -527,6 +537,32 @@ private void extractIntentFields(Intent intent) {
         onSharedIntent(intent);
     }
 
+    private void openLoyalityCardFromWidget(int itemId)
+    {
+        DBHelper.updateLoyaltyCardLastUsed(mDatabase, itemId);
+        Intent clickIntent = new Intent(this, LoyaltyCardViewActivity.class);
+        clickIntent.setAction("");
+        final Bundle b = new Bundle();
+        b.putInt(LoyaltyCardViewActivity.BUNDLE_ID, itemId);
+
+        ArrayList<Integer> cardList = new ArrayList<>();
+        for (int i = 0; i < mAdapter.getItemCount(); i++) {
+            cardList.add(mAdapter.getCard(i).id);
+        }
+
+        b.putIntegerArrayList(LoyaltyCardViewActivity.BUNDLE_CARDLIST, cardList);
+        clickIntent.putExtras(b);
+        startActivity(clickIntent);
+    }
+    private void onOpenItemExtra(Intent intent)
+    {
+        if (intent != null && (intent.hasExtra("item_id"))) {
+            int itemId = intent.getIntExtra("item_id", -1);
```

Comment:
I think this should be using `LoyaltyCardViewActivity.BUNDLE_ID`, as that refers to the card ID.